core/client/fs/src/main/java/alluxio/client/file/options/OutStreamOptions.java

```diff
@@ -63,9 +64,11 @@ private OutStreamOptions() {
     try {
       mLocationPolicy = CommonUtils.createNewClassInstance(
           Configuration.<FileWriteLocationPolicy>getClass(
-              PropertyKey.USER_FILE_WRITE_LOCATION_POLICY), new Class[] {}, new Object[] {});
+              PropertyKey.USER_FILE_WRITE_LOCATION_POLICY), new Class[]{}, new Object[]{});
+    } catch (InvocationTargetException e) {
+      throw new RuntimeException(e.getCause());
     } catch (Exception e) {
-      throw Throwables.propagate(e);
+      throw new RuntimeException(e);
```

Comment:
Should we use a utility method for this try catch logic?

Reply:
I moved the logic inside `createNewClassInstance` and changed it to throw only unchecked exceptions. There were a couple of places where we would try to create a user-defined class and fall back on another class if we fail to instantiate the configured class. I've changed those sites to propagate the exception instead. It's better to crash at startup than to let the user go on thinking we're using their implementations when we really aren't.

Reply:
Thanks, this is much cleaner.